pinecone/control/index_host_store.py

```diff
@@ -0,0 +1,39 @@
+from pinecone.config import Config
+from pinecone.core.client.api.index_operations_api import IndexOperationsApi
+
+
+class SingletonMeta(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            instance = super().__call__(*args, **kwargs)
+            cls._instances[cls] = instance
+        return cls._instances[cls]
+
+
+class IndexHostStore(metaclass=SingletonMeta):
```

Comment:
I confess, I had help from ChatGPT on this. I still need to add tests to verify the singleton is working as expected.